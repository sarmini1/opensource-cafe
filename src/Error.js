import React from "react";

/**Error
 *
 * Props:
 * - error: string
 *
 * State:
 * - none
 *
 * App, NewItemForm --> Error
 */
function Error({ error }) {
  return (
    <div className="Error alert alert-danger">
      <h3>An error occurred:</h3>
      {error}
    </div>
  );
}

export default Error;