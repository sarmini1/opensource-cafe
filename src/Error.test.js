import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";

import Error from "./Error";

it('mounts without crashing', function () {
  render(<MemoryRouter><Error error={"test error"}/></MemoryRouter>);
});

it("matches snapshot", function () {
  const { container } = render(
    <MemoryRouter><Error error={"test error"}/></MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});

it("displays Error component with appropriate HTML", function () {
  const { debug, container } = render(
    <MemoryRouter><Error error={"test error"}/></MemoryRouter>
  );

  debug(container);
  const error = container.querySelector(".Error");
  expect(error).toContainHTML(
    '<h3>An error occurred:</h3>'
  )
  expect(error).toContainHTML(
    'test error'
  )
});