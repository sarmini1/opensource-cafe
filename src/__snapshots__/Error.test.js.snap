// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Error alert alert-danger"
  >
    <h3>
      An error occurred:
    </h3>
    test error
  </div>
</div>
`;
